src/state.py

```diff
@@ -254,17 +300,19 @@ def from_charm(cls, charm: ops.CharmBase) -> "State":
             CharmRelationDataInvalidError: if invalid relation data was received.
             CharmIllegalNumUnitsError: if more than 1 unit of Jenkins charm is deployed.
         """
-        time_range_str = charm.config.get("restart-time-range")
-        if time_range_str:
-            try:
+        try:
+            remoting_config = RemotingConfig.from_config(config=charm.config)
```

Comment:
I feel like the except blocks could benefit from a refactoring. I see lots of `ValidationError` that only differ in the log message, which is probably not needed as the invalid config can be identified by means of the trace